packages/mangrove-arbitrage/src/MgvArbitrage.sol

```diff
@@ -1,55 +1,82 @@
-// SPDX-License-Identifier: BSD-2-Clause
+// SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.13;
 
-import {ISwapRouter} from "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
-import {IMangrove} from "mgv_src/IMangrove.sol";
-import {IERC20} from "mgv_src/IERC20.sol";
-import {AccessControlled} from "mgv_src/strategies/utils/AccessControlled.sol";
-import {TransferLib} from "mgv_src/strategies/utils/TransferLib.sol";
+import {IMangrove} from "@mgv/src/IMangrove.sol";
+import {IERC20, OLKey} from "@mgv/src/core/MgvLib.sol";
+import {AccessControlled} from "@mgv-strats/src/strategies/utils/AccessControlled.sol";
+import {TransferLib} from "@mgv/lib/TransferLib.sol";
+import {IUniswapV3Pool} from "@uniswap/v3/lib/contracts/interfaces/IUniswapV3Pool.sol";
+import {IUniswapV3SwapCallback} from "@uniswap/v3/lib/contracts/interfaces/callback/IUniswapV3SwapCallback.sol";
+import {FullMath} from "./FullMath.sol";
 
-/// @param offerId The offer to be arbitraged
-/// @param takerWantsToken The token the taker wants
-/// @param takerWants The amount the taker wants
 /// @param takerGivesToken The token the taker gives
-/// @param takerGives The amount the taker gives
-/// @param fee The fee used for the swap on Uniswap
-/// @param minGain The minimum amount the arbitrage needs to gain
+/// @param takerWantsToken The token the taker wants
+/// @param pool The Uniswap pool
+/// @param minimumGain The minimum amount the arbitrage needs to gain
 struct ArbParams {
-  uint offerId;
-  address takerWantsToken;
-  uint takerWants;
-  address takerGivesToken;
-  uint takerGives;
-  uint24 fee;
-  uint minGain;
+  IERC20 takerGivesToken;
+  IERC20 takerWantsToken;
+  uint tickSpacing;
+  IUniswapV3Pool pool;
+  uint160 minimumGain; // if zero then ignored
 }
 
-contract MgvArbitrage is AccessControlled {
+contract MgvArbitrage is AccessControlled, IUniswapV3SwapCallback {
+  /// copied from v3-core repo to interact with low level swap function of uniwap
+  uint160 internal constant MIN_SQRT_RATIO = 4295128739;
+  uint160 internal constant MAX_SQRT_RATIO = 1461446703485210103287273052203988822378723970342;
+
   IMangrove public mgv;
-  address public arbitrager;
+
+  bool isArbitraging;
+
+  mapping(address => bool) public pools;
```

Comment:
Sorry but can you netspec these, it's not very clear what the address is referring to at this point.